stream/src/main/java/org/apache/kafka/clients/processor/Stamped.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.processor;
+
+public class Stamped<V> implements Comparable {
+
+    public final V value;
+    public final long timestamp;
+
+    public Stamped(V value, long timestamp) {
+        this.value = value;
+        this.timestamp = timestamp;
+    }
+
+    public int compareTo(Object other) {
```

Comment:
OK makes sense, however after a second thought I feel by "consistency" we want:

1) if compareTo() returns none-zero, equals() should return false;
2) if equals() returns true, compareTo() should return zero.

but:

3) if equals() returns false, compareTo() does not necessarily returns none-zero.
4) if compareTo() returns zero, equals() does not necessarily returns true.

If we enforce 3) and 4) as well, it means sorted set / map will not allow two records who are comparably same to each other as the docs stated; but for our case, we actually want two  Stamped objects with the same timestamp to still be stored at the same time as keys if we ever want to do so.